packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/representation/services/RepresentationSearchService.java

```diff
@@ -68,4 +83,34 @@ private Optional<IRepresentation> toRepresentation(String content) {
         }
         return optionalRepresentation;
     }
+
+    private Optional<IRepresentation> migrateContent(String content) {
+        try {
+            JsonNode rootJsonNode = this.objectMapper.readTree(content);
+            ObjectNode rootObjectNode = (ObjectNode) rootJsonNode;
+            var id = rootJsonNode.get("id").asText();
+            var representationData = this.representationDataSearchService.findById(UUID.fromString(id));
+
+            if (representationData.isPresent()) {
+                var lastMigrationPerformed = representationData.get().getLastMigrationPerformed();
+                var migrationVersion = representationData.get().getMigrationVersion();
+                var migrationService = new MigrationService(migrationParticipants, new MigrationData(lastMigrationPerformed, migrationVersion), rootObjectNode);
+                migrationService.parseProperties(rootObjectNode, this.objectMapper);
+
+                var optionalRepresentation = this.toRepresentation(rootObjectNode.toString());
+
+                if (optionalRepresentation.isPresent()) {
+                    migrationService.getLastMigrationParticipant().ifPresent(migrationParticipant -> {
+                        this.representationDataUpdateService.updateMigrationData(UUID.fromString(id), migrationParticipant.getClass().getSimpleName(), migrationParticipant.getVersion());
```

Comment:
Technically here you are writing things in the database which is exactly what we don't want from the search service. You should only modify things in memory here